examples/ios/objc/Puzzle/Controllers/RLMViewController.m

```diff
@@ -0,0 +1,249 @@
+    self.puzzleView.delegate = self;
+    [self.view addSubview:self.puzzleView];
+
+    self.startView = [[RLMStartView alloc] initWithFrame:self.view.frame];
+    [self.view addSubview:self.startView];
+    
+    __weak typeof(self) weakSelf = self;
+    self.startView.connectButtonTapped = ^{
+        [weakSelf connectToServer];
+    };
+    
+    UITapGestureRecognizer *tapRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(resetGestureRecognized:)];
+    tapRecognizer.numberOfTouchesRequired = 3;
+    tapRecognizer.numberOfTapsRequired = 3;
+    [self.view addGestureRecognizer:tapRecognizer];
+}
+
+- (BOOL)prefersStatusBarHidden
+{
+    return YES;
+}
+
+#pragma mark - Puzzle State Management -
+- (void)connectToServer
+{
+    self.startView.loading = YES;
+    
+    NSURL *authURL = [NSURL URLWithString:[NSString stringWithFormat:@"http://%@:9080", self.startView.hostName]];
+    
+    RLMSyncCredential *credential = [RLMSyncCredential credentialWithUsername:self.startView.userName password:self.startView.password register:NO];
+    [RLMSyncUser logInWithCredential:credential authServerURL:authURL onCompletion:^(RLMSyncUser *user, NSError *error) {
+        if (error) {
+            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"An Error Ocurred" message:error.localizedDescription preferredStyle:UIAlertControllerStyleAlert];
+            [alertController addAction:[UIAlertAction actionWithTitle:@"Okay" style:UIAlertActionStyleDefault handler:nil]];
+            self.startView.loading = NO;
+            [self presentViewController:alertController animated:YES completion:nil];
+            return;
+        }
+        
+        NSURL *syncURL = [NSURL URLWithString:[NSString stringWithFormat:@"realm://%@:9080/~/Puzzle", self.startView.hostName]];
+        
+        RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
+        [configuration setSyncConfiguration:[[RLMSyncConfiguration alloc] initWithUser:user realmURL:syncURL]];
+        [RLMRealmConfiguration setDefaultConfiguration:configuration];
+        
+        RLMRealm *realm = [RLMRealm defaultRealm];
+        RLMPuzzle *puzzle = [RLMPuzzle objectForPrimaryKey:kRLMPuzzleUUID];
+        if (!puzzle) {
+            RLMPuzzle *puzzle = [[RLMPuzzle alloc] init];
+            puzzle.uuid = kRLMPuzzleUUID;
+            [realm transactionWithBlock:^{
+                [realm addObject:puzzle];
+            }];
+        }
+        
+        [self startPuzzle];
+    }];
+    
+}
+
+- (void)startPuzzle
+{
+    //Create the over-arching puzzle object
+    RLMPuzzle *newPuzzle = [RLMPuzzle objectForPrimaryKey:kRLMPuzzleUUID];
+    self.currentPuzzleID = newPuzzle.uuid;
+    
+    BOOL firstTime = NO;
+    if (newPuzzle.pieces.count == 0) {
+        //Create a data point for each puzzle piece
+        NSMutableArray *puzzlePieces = [NSMutableArray array];
```

Comment:
Why not merge the code to create the initial puzzle pieces with the code that creates the initial puzzle?